lucupy/minimodel/observation.py

```diff
@@ -116,78 +157,88 @@ class Observation:
         """
         return self.program_used() + self.partner_used()
 
     def required_resources(self) -> FrozenSet[Resource]:
         """
-        The required resources for an observation based on the sequence's needs.
+        Returns:
+            The required resources for an observation based on the sequence's needs.
         """
         # TODO: For now, we do not return guiding keys amongst the resources.
         # return frozenset(self.guiding.keys() | {r for a in self.sequence for r in a.resources})
         return frozenset((r for a in self.sequence for r in a.resources))
 
     def instrument(self) -> Optional[Resource]:
         """
-        Returns a resource that is an instrument, if one exists.
-        There should be only one.
+        Returns: 
+            A resource that is an instrument, if one exists. There should be only one.
         """
         return next(filter(lambda r: ObservatoryProperties.is_instrument(r),
                            self.required_resources()), None)
 
     def wavelengths(self) -> FrozenSet[float]:
         """
-        The set of wavelengths included in the sequence.
+        Returns:
+            The set of wavelengths included in the sequence.
         """
         return frozenset((w for c in self.sequence for w in c.wavelengths))
 
     def constraints(self) -> FrozenSet[Constraints]:
         """
-        A set of the constraints required by the observation.
-        In the case of an observation, this is just the (optional) constraints.
+        Returns:
+            A set of the constraints required by the observation.
+            In the case of an observation, this is just the (optional) constraints.
         """
         return frozenset([self.constraints] if self.constraints is not None else [])
 
     def program_used(self) -> timedelta:
-        """
-        We roll this information up from the atoms as it will be calculated
-        during the GreedyMax algorithm. Note that it is also available directly
-        from the OCS, which is used to populate the time allocation.
+        """We roll this information up from the atoms as it will be calculated
+            during the GreedyMax algorithm. Note that it is also available directly
+            from the OCS, which is used to populate the time allocation.
+        Returns:
+            (timedelta): With the time program used.
         """
         return sum((atom.prog_time for atom in self.sequence), start=timedelta())
 
     def partner_used(self) -> timedelta:
-        """
-        We roll this information up from the atoms as it will be calculated
+        """We roll this information up from the atoms as it will be calculated
         during the GreedyMax algorithm. Note that it is also available directly
         from the OCS, which is used to populate the time allocation.
+
+        Returns:
+            (timedelta): With the time program used.
```

Comment:
We should probably change GreedyMax in the docs to Optimizer everywhere.

The definition of the return value is a bit confusing? It is just the sum of the allocated partner time that has been used.